### ISSUE TITLE
Not evaluable expression should be treated as null when using skipValue

### ISSUE THREAD

**Issue by mapedraza** [telefonicaid/iotagent-node-lib#1440]
### IoT Agent Node Lib version the issue has been seen with

Latest (3.3.0-next)

### Bound or port used (API interaction)

Other

### NGSI version

NGSIv2

### Are you running a container?

Yes, I am using a contaner (Docker, Kubernetes...)

### Image type

None

### Expected behaviour you didn't see

When an active attribute expression involving a measurement (`"expression":"t*2"`)is evaluated, if missing parameter (I.E: when a device sends `{"m":3}` the resulting entity attribute is `null`. This is the expected.

When defining active attribute expression as constant `null` (`"expression":null`), the resulting entity attribute is `null`, and `"skipValue":null`, the attribute is not updated by the IoTA. This is the expected value

When, for a given active attribute, with `"skipValue":null` and an expression `"expression":"t*2"`, a measurement without that parameter (`{"m":3}`) arrives, the attribute is written with a `null` value. This is **not** the expected result. It is expected that IoTA does not update the value. 

There is a workaround for this, in that case defining the expression like `t?t*2:null`, which will force to null the value. 

Probably, this is because internally, the result of an expression that does not have all the parameters available is an error that, in a subsequent stage, is transformed to a null. And the comparison with skipValue, in order to avoid persisting it into the CB is done in between.

**Comment by AlvaroVega** [comment]
https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md
 By default if skipValue is not defined then is considered as null 

So this case

`When an active attribute expression involving a measurement ("expression":"t*2")is evaluated, if missing parameter (I.E: when a device sends {"m":3} the resulting entity attribute is null. This is the expected.`

is the same that
`When, for a given active attribute, with "skipValue":null and an expression "expression":"t*2", a measurement without that parameter ({"m":3}) arrives, the attribute is written with a null value. `

**Comment by mrutid** [comment]
"When, for a given active attribute, with "skipValue":null and an expression "expression":"t*2", a measurement without that parameter ({"m":3}) arrives, the attribute is written with a null value. This is not the expected result. It is expected that IoTA should not update the value."

In JEXL null*2 == null no exception or error.  are we sure about this behaviour?

It's not the same than "nojson"|jsonparse (Error: SyntaxError: Unexpected end of JSON input) in which case we convey to return null value also.

**Comment by AlvaroVega** [comment]
in this case: 
 ```javascript 
res === attr.skipValue```
computation result is false since res is a number (null *2 -> 0) and attrSkipValue is null

```
 js
Welcome to Node.js v18.13.0.
Type ".help" for more information.
> null * 2
0
```

**Comment by fgalan** [comment]
PR https://github.com/telefonicaid/iotagent-node-lib/pull/1441 has been merged.

@mapedraza maybe it would be a good idea checking the fix before closing this issue (I'm relaunching IOTA-JSON, IOTA-UL and IOTA Manager builds at dockerhub right now, to get the latest from IOTA library code).

**Comment by AlvaroVega** [comment]
The following cases should be tested with iotagent after https://github.com/telefonicaid/iotagent-node-lib/pull/1441
```
{
a:v,
b:v*3,
c:v==null,
d:v?"soy null":"no soy null",
e:v==null?"soy null":"no soy null",
f:(v*3)?"soy null":"no soy null",
g:v == undefined
}
```

**Comment by mapedraza** [comment]
I am going to open a new PR adding a test for cases proposed by @AlvaroVega. After that, if everything works well, we can close the issue as proposed by @fgalan

**Comment by AlvaroVega** [comment]
> The following cases should be tested with iotagent after #1441
> 
> ```
> {
> a:v,
> b:v*3,
> c:v==null,
> d:v?"soy null":"no soy null",
> e:v==null?"soy null":"no soy null",
> f:(v*3)?"soy null":"no soy null",
> g:v == undefined
> }
> ```

I believe that these cases are currently covered by https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/ngsiv2/expressions/jexlBasedTransformations-test.js

**Comment by mapedraza** [comment]
> I believe that these cases are currently covered by https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/ngsiv2/expressions/jexlBasedTransformations-test.js

Added here: https://github.com/telefonicaid/iotagent-node-lib/pull/1471
